src/main/groovy/org/ajoberstar/grgit/auth/AuthConfig.java

```diff
@@ -0,0 +1,266 @@
+      return null;
+    }
+  }
+
+  /**
+   * Gets the path to your SSH private key to use during authentication reflecting the value set in
+   * the system properties.
+   *
+   * @return the path to the SSH key, if set, otherwise {@code null}
+   */
+  public String getSshPrivateKeyPath() {
+    return props.get(SSH_PRIVATE_KEY_OPTION);
+  }
+
+  /**
+   * Gets the passphrase for your SSH private key to use during authentication reflecting the value
+   * set in the system properties.
+   *
+   * @return the passphrase of the SSH key, if set, otherwise {@code null}
+   */
+  public String getSshPassphrase() {
+    return props.get(SSH_PASSPHRASE_OPTION);
+  }
+
+  /**
+   * Gets session config override for SSH session that is used underneath by JGit
+   *
+   * @return map with configuration or empty if nothing was specified in system property
+   */
+  public Map<String, String> getSessionConfig() {
+    return props
+        .entrySet()
+        .stream()
+        .filter(entry -> entry.getKey().startsWith(SSH_SESSION_CONFIG_OPTION_PREFIX))
+        .collect(
+            Collectors.toMap(
+                entry -> entry.getKey().substring(SSH_SESSION_CONFIG_OPTION_PREFIX.length()),
+                Map.Entry::getValue));
+  }
+
+  /**
+   * Factory method to construct an authentication configuration from the given properties and
+   * environment.
+   *
+   * @param properties the properties to use in this configuration
+   * @param env the environment vars to use in this configuration
+   * @return the constructed configuration
+   * @throws GrgitException if force is set to an invalid option
+   */
+  public static AuthConfig fromMap(Map<String, String> props) {
+    return new AuthConfig(props, Collections.emptyMap());
+  }
+
+  /**
+   * Factory method to construct an authentication configuration from the given properties and
+   * environment.
+   *
+   * @param properties the properties to use in this configuration
+   * @param env the environment vars to use in this configuration
+   * @return the constructed configuration
+   * @throws GrgitException if force is set to an invalid option
+   */
+  public static AuthConfig fromMap(Map<String, String> props, Map<String, String> env) {
+    return new AuthConfig(props, env);
+  }
+
+  /**
+   * Factory method to construct an authentication configuration from the current system properties
+   * and environment variables.
+   *
+   * @return the constructed configuration
+   * @throws GrgitException if force is set to an invalid option
+   */
+  public static AuthConfig fromSystem() {
+    Map<String, String> sysProps =
+        System.getProperties()
+            .entrySet()
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> entry.getKey().toString(), entry -> entry.getValue().toString()));
+    return fromMap(sysProps, System.getenv());
+  }
+
+  /** Available authentication options. */
+  public static enum Option {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant "static" qualifier. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.ajoberstar.com/coding_rules#rule_key=squid%3AS2786)